src/roomba_robot/tasks/botgal_and_lib/botgal_task.c

```diff
@@ -0,0 +1,190 @@
+    printf("%d,%d,%s", get_create_rcliff(), 0, "right cliff \n");
+    printf("%d,%d,%s", get_create_rlightbump(), 0, "right light bumper \n");
+    printf("%d,%d,%s", get_create_rbump(), 0, "right bumper \n");
+    printf("%d,%d,%s", get_create_rflightbump(), 0, "right light front bumper \n");
+    printf("%d,%d,%s", get_create_rfcliff(), 0, "right cliff \n");
+    printf("%d,%d,%s", get_create_rclightbump(), 0, "right cliff light bumper \n");
+    printf("%d,%d,%s", get_create_cwdrop(), 0, "front drop wheel \n");
+    printf("%d,%d,%s", get_create_lclightbump(), 0, "left cliff light bumper \n");
+    printf("%d,%d,%s", get_create_lfcliff(), 0, "left front cliff \n");
+    printf("%d,%d,%s", get_create_lflightbump(), 0, "left light bumper \n");
+    printf("%d,%d,%s", get_create_lbump(), 0, "left bumper \n");
+    printf("%d,%d,%s", get_create_llightbump(), 0, "left light bumper \n");
+    printf("%d,%d,%s", get_create_lcliff(), 0, "left cliff \n");
+    printf("%d,%d,%s", get_create_lwdrop(), 0, "left wheel drop \n");
+    printf("%d,%d,%s", get_create_distance(), 0, "total distance\n");
+    printf("%d,%d,%s", get_create_total_angle(), 0, "total angle \n");
+}
+
+bool setupcamera() {
+    if (camera_open() == 0 || camera_update() == 0) return false;
+    return true;
+}
+
+void setuprobot(bool debug, bool camera) {
+    create_connect();
+    setupsensordata();
+#ifdef DEBUG
+    printsensordata();
+#endif
+    if (camera) {
+        if (setupcamera()) printf("%s", "Camera initialized");
+        else printf("%s", "Camera Error");
+    }
+    enable_servos();
+}
+
+void shutdownrobot(bool camera) {
+    if (camera) {
+        camera_close();
+    }
+    disable_servos();
+    ao();
+    create_stop();
+    create_disconnect();
+}
+
+void drivestraight(int speed, enum direction{forward,backward}; direction, int distance,bool watch_for_touch) { // distance in mm
+    if (direction == forward) {
+        set_create_distance(0);
+        while (get_create_distance() <= (distance)) {
+            create_drive_direct(speed, speed);
+            if (watch_for_touch) {
+                if (get_create_lbump() || get_create_rbump() == true) {
+                    set_create_distance(distance);
+                }
+            }
+        }
+    } else {
+        set_create_distance(0);
+        while ((-1 * get_create_distance()) <= (distance)) {
+            create_drive_direct(speed * -1, speed * -1);
+        }
+    }
+}
+
+void turn(int speed, int deg) {
+    // deg - turn to right  + turn to left
+    set_create_distance(0);
+    set_create_total_angle(0);
+    if (deg > 0) {
+        while (get_create_total_angle() <= deg) {
+            create_drive_direct(speed * -1, speed);
+        }
+    } else {
+        while (get_create_total_angle() >= deg) {
+            create_drive_direct(speed, speed * -1);
+        }
+    }
+}
+
+void wasd() { // function to control robot via webinterface
+    bool exit;
```

Comment:
Instead of:

```c
bool exit;
exit = true;
```

you can just write 

```c
bool exit = true;
```

Also it doesnt make much sense to say "while exit is true"... instead make ``exit = false`` and write "while exit !false"